packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -984,6 +1049,59 @@ class MediaQuery extends InheritedModel<_MediaQueryAspect> {
     );
   }
 
+  /// Wraps the `child` in a [MediaQuery] with its [MediaQueryData.textScaler]
+  /// set to [TextScaler.noScaling].
+  ///
+  /// The returned widget must be inserted in a widget tree below an existing
+  /// [MediaQuery] widget.
+  ///
+  /// This can be used to prevent, for example, icon fonts from scaling as the
+  /// user adjusts the platform's text scaling value.
+  static Widget disableTextScaling({
+    Key? key,
+    required Widget child,
+  }) {
+    return Builder(
+      key: key,
+      builder: (BuildContext context) {
+        assert(debugCheckHasMediaQuery(context));
+        return MediaQuery(
+          data: of(context).copyWith(textScaler: TextScaler.noScaling, textScaleFactor: 1.0),
```

Comment:
lol, the stand-alone `of` looks somewhat funny. Maybe just spell it out as `MediaQuery.of(context)` to be explicit? I believe the other existing convenience statics on MediaQuery also do it that way...

(here and elsewhere)

Reply:
done